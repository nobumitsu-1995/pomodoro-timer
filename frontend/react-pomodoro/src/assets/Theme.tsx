import React from 'react'
import { ThemeProvider } from 'styled-components'

type Props = {
  children: React.ReactNode
}

const Theme: React.FC<Props> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const theme = {
  color: {
    black: '#000',
    white: '#fff',
    red: 'red',
    blue: 'blue',
    green: 'green',
  },
  size: {
    small: '1.0rem',
    normal: '1.2rem',
    big: '2.0rem',
  },
} as const

export default Theme
