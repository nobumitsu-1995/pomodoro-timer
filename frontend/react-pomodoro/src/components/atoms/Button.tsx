import React from 'react'
import styled from 'styled-components/macro'

type ButtonProps = Omit<JSX.IntrinsicElements['button'], 'ref'>

type Props = ButtonProps & {
  /** 描画する要素 */
  children: React.ReactNode | string
  /** font-sizeの値 */
  fontsize?: string
  /** width, heightの値 */
  size?: string
}

/** ニューモフィズムButtonコンポーネント */
const Button: React.FC<Props> = (props) => {
  const { children, fontsize, size, ...buttonProps } = props

  return (
    <StyledButton
      type='button'
      {...buttonProps}
      css={`
        height: ${size || '100%'};
        width: ${size || '100%'};
        font-size: ${fontsize || '1.2rem'};
      `}
    >
      {children}
    </StyledButton>
  )
}

export default Button

const StyledButton = styled.button`
  padding: 10px;
  background: #ebebeb;
  box-shadow: 10px 10px 20px #c8c8c8, -10px -10px 20px #ffffff;
  border-radius: 50%;
  text-align: center;
  color: ${(props) => props.theme.color.black};
  transition: box-shadow ease-out 0.13s;

  &:hover {
    box-shadow: 10px 10px 20px #bcbcbc, -10px -10px 20px #ffffff;
  }

  &:active,
  &:disabled {
    box-shadow: 10px 10px 20px #e2e2e2, -10px -10px 20px #f4f4f4;
  }
`
