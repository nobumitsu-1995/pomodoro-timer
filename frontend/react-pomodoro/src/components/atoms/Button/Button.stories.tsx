import React from 'react'
import { Meta, Story } from '@storybook/react'

import Button, { Props } from './Button'

export default {
  title: 'atoms/Button',
  component: Button,
  argTypes: {},
} as Meta

const Template: Story<Props> = (args) => <Button {...args} />

export const Default = Template.bind({})

Default.args = {
  children: 'Button',
  fontsize: '',
  size: '100px',
  borderradius: '',
}
