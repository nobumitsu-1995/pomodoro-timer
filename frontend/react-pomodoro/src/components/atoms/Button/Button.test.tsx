/* eslint-disable react/react-in-jsx-scope */
import * as stories from './Button.stories'
import { composeStories } from '@storybook/react'
import { render } from '@testing-library/react'
import Theme from 'src/assets/styles/Theme'

describe('snapshot', () => {
  Object.entries(composeStories(stories)).forEach(([name, Component]) => {
    describe(name, () => {
      it('should render correctly', () => {
        const tree = render(
          <Theme>
            <Component {...Component.args} />
          </Theme>
        ).asFragment()
        expect(tree).toMatchSnapshot()
      })
    })
  })
})
