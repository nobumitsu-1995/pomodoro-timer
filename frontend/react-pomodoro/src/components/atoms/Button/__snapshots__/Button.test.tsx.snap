// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Default should render correctly 1`] = `
<DocumentFragment>
  <button
    class="Button__StyledButton-sc-mes1hd-0 Button___StyledStyledButton-sc-mes1hd-1 fxBFHR bbtMuN"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
